syntax = "proto3";

package rpc;

// The greeting service definition.
service OpaqueRPC {
  rpc relayGenerateReport (RARequest) returns (RAReply) {}
  rpc relayFinishAttestation (KeyRequest) returns (KeyReply) {}
  rpc relayQuery (QueryRequest) returns (QueryReply) {}
}

// The response message containing the initial RARequest from the client
message RARequest {
  string name = 1;
}

message RAReply {
  bool success = 1;
  repeated bytes report = 2;
}

message KeyRequest {
  bool nonNull = 1;
  string name = 2;
  repeated bytes key = 3;
}

message KeyReply {
  bool success = 1;
}

message QueryRequest {
  bool encryptedCreate = 1;
  string sqlQuery = 2;
}

message QueryReply {
  bool success = 1;

  // The data is returned in rows as strings
  repeated string data = 2;
}
